### ISSUE TITLE
Thumbnail generation issue for layer in non EPSG:4326 projection

### ISSUE THREAD

**Issue by afabiani** [GeoNode/geonode#3593]
I’ve encountered an issue where I’m getting GeoNode to generate layer thumbnails as part of the updatelayers command. The problem I have is that when I configure the layer in GeoServer in order to get a valid thumbnail produced I then lose the ability to filter by extent for that layer in GeoNode. I believe this is because the layer in question is in a UK projection system and not EPSG:4326.

In order to get a valid thumbnail produced I have to set the following properties in GeoServer:

![image](https://user-images.githubusercontent.com/1278021/35745728-9cade094-0844-11e8-8566-319478ae26b7.png)

However with these details set when I filter by extent in GeoNode using either the API or web interface the layer is no longer returned (example filter: http://<geonode_host>/layers/?limit=100&offset=0&extent=-6.899414062500001,48.07807894349862,4.658203125000001,56.145549500679074)

I think this is because the extent being provided via the filter is in EPSG:4326 (as this is what GeoNode supports) but the lat/lon Bounding Box in GeoServer is in EPSG:27700. If I change the Lat/Lon Bounding Box to be EPSG:4326 coordinates in GeoServer then I get a blank white image generated for the thumbnail and the extent filter starts working again.

I’ve tried all the combinations and SRS Handling, Native Bounding Box and Lat/Lon Bounding Box but I cannot get a valid thumbnail and the extent filter to work at the same time.

Is what I’m trying to do possible? Thanks in advance for any feedback!

Thanks,
Pete

**Comment by marlowp** [comment]
I've resolved this issue temporarily for our project by patching geonode (not ideal).

I've updated the geonode.geoserver.helpers file so that it no longer appends the BBOX parameter to the "wms/reflect?" call. This way geoserver is in charge of determining the BBOX for the layer and produces a BBOX in the correct projection.

So basically I've done the following:
```
# Add the bbox param only if the bbox is different to [None, None,
# None, None]
check_bbox = False
#if None not in instance.bbox:
#    params['bbox'] = instance.bbox_string
#    check_bbox = True
```

**Comment by afabiani** [comment]
@marlowp I see to issues here:

1. On GeoServer the Lon/Lat BBOX is wrong
2. Don't know exactly which version of GeoNode/GeoServer you are currently using, but on GeoNode 2.7.x+ with GeoServer 2.12.x the correct solution would be simply to add the CRS parameter to the request e.g.:

```
    check_bbox = False
    if None not in instance.bbox:
        params['bbox'] = instance.bbox_string
        params['crs'] = instance.srid
        check_bbox = True
```

I tested this locally by using your same projection (see the sample attached). And it works.
I'm going to create a PR on master branch by adding crs param to the getMap request as specified above though

[map_UK.zip](https://github.com/GeoNode/geonode/files/1727632/map_UK.zip)

**Comment by marlowp** [comment]
Thanks for the reply @afabiani 

Your fix is much cleaner, I'm using GeoNode 2.6.3 and GeoServer 2.12-SNAPSHOT and can confirm that passing the CRS parameter to the GeoServer call also resolves the issue.
